src/components/HomePage/MainView.spec.js

```diff
@@ -24,10 +26,12 @@ describe("Main View", () => {
         showSearchResults = jest.fn();
 
         wrapper = mountWithStore(
-            <MainView
-                scrollToProductDescription={scrollToProductDescription}
-                showSearchResults={showSearchResults}
-            />,
+            <MemoryRouter initialEntries={["/"]}>
+                <MainView
+                    scrollToProductDescription={scrollToProductDescription}
+                    showSearchResults={showSearchResults}
+                />
+            </MemoryRouter>,
```

Comment:
Do you mind explaining the use of `MemoryRouter`? I'm not familiar

Reply:
We need to have a Router, no matter what type it is, around every component that uses the AdvancedSearch because of the use of `useLocation` and `useHistory`: in Jest these do not work because JSDOM does not provide values for `location` and `history`, things which are handled by a Router. Besides that, a `MemoryRouter` lets us input directly the history stack, which can be useful for the tests.